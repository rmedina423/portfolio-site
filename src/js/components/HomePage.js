import React from 'react';
import uuidv1 from 'uuid/v1';
import chunk from 'lodash.chunk';
import Tile from './Tile';
import collectionOfPhotos from '../collectionOfPhotos';

export default function HomePage() {
  const numberOfColumns = getNumberOfColumns(window.innerWidth);
  const collectionSets = getCollectionSets(collectionOfPhotos, numberOfColumns);

  return (
    <div className="tile-container">
      {[...Array(numberOfColumns)].map((_, i) => (
        <div className="tile-column" key={uuidv1()}>
          {collectionSets[i].map(collectionProps => (
            <Tile {...collectionProps} key={collectionProps.title} />
          ))}
        </div>
      ))}
    </div>
  );
}

function getCollectionSets(collection, numberOfColumns) {
  const numberOfTilesPerColumn = Math.floor(collection.length / numberOfColumns);
  const collectionSets = chunk(collection, numberOfTilesPerColumn);
  return distributeExcessCollection(collectionSets, numberOfColumns);
}

function distributeExcessCollection(collectionSets, numberOfColumns) {
  if (collectionSets.length !== numberOfColumns) {
    const excessCollection = collectionSets
      .splice(numberOfColumns, collectionSets.length - numberOfColumns)
      .reduce((a, b) => a.concat(b), []);

    excessCollection.forEach((collection, i) => {
      collectionSets[i].push(collection);
    });
  }

  return collectionSets;
}

function getNumberOfColumns(windowInnerWidth) {
  if (windowInnerWidth >= 1515) return 6;
  if (windowInnerWidth >= 1265) return 5;
  if (windowInnerWidth >= 1015) return 4;
  if (windowInnerWidth >= 765) return 3;
  if (windowInnerWidth >= 515) return 2;
  if (windowInnerWidth >= 0) return 1;
}
