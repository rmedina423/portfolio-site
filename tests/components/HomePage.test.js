import React from 'react';
import { shallow } from 'enzyme';

import HomePage from '../../src/js/components/HomePage';

describe('HomePage Component', function () {
  beforeEach(() => {
    this.homePage = shallow(<HomePage />);
  });

  test('should match snapshot', () => {
    expect(this.homePage.getElement()).toMatchSnapshot();
  });

  test('on Masonry `onLayoutComplete`, state.layoutHidden is set to false if not already set to false', () => {
    const onLayoutComplete = this.homePage.find('MasonryComponent').prop('onLayoutComplete');
    expect(this.homePage.state('layoutHidden')).toEqual(true);
    onLayoutComplete();
    expect(this.homePage.state('layoutHidden')).toEqual(false);
  });

  test('on Masonry `onLayoutComplete`, state.layoutHidden is not set if already set to false', () => {
    this.homePage.setState({ layoutHidden: false });
    const onLayoutComplete = this.homePage.find('MasonryComponent').prop('onLayoutComplete');
    expect(this.homePage.state('layoutHidden')).toEqual(false);
    onLayoutComplete();
    expect(this.homePage.state('layoutHidden')).toEqual(false);
  });
});