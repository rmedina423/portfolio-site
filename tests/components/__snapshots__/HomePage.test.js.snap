// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should match snapshot 1`] = `
<div
  className="tile-container"
>
  <div
    className="tile-column"
  >
    <Tile
      imgName="vegas-2016.jpg"
      link="/coming-soon"
      title="Vegas 2016"
    />
    <Tile
      imgName="denver-2016.jpg"
      link="/coming-soon"
      title="Denver 2016"
    />
    <Tile
      imgName="tonto-national-bridge-2016.jpg"
      link="/coming-soon"
      title="Tonto National Bridge 2016"
    />
    <Tile
      imgName="jerome-2017.jpg"
      link="/coming-soon"
      title="Jerome 2017"
    />
  </div>
  <div
    className="tile-column"
  >
    <Tile
      imgName="valentines-2017.jpg"
      link="/coming-soon"
      title="Valentines 2017"
    />
    <Tile
      imgName="thanksgiving-2016.jpg"
      link="/coming-soon"
      title="Thanksgiving 2016"
    />
    <Tile
      imgName="park-photoshoot-2017.jpg"
      link="/coming-soon"
      title="Park Photoshoot 2017"
    />
  </div>
  <div
    className="tile-column"
  >
    <Tile
      imgName="sedona-2016.jpg"
      link="/coming-soon"
      title="Sedona 2016"
    />
    <Tile
      imgName="flagstaff-2016.jpg"
      link="/coming-soon"
      title="Flagstaff 2016"
    />
    <Tile
      imgName="san-diego-2016.jpg"
      link="/coming-soon"
      title="San Diego 2016"
    />
  </div>
  <div
    className="tile-column"
  >
    <Tile
      imgName="denver-2017.jpg"
      link="/coming-soon"
      title="Denver 2017"
    />
    <Tile
      imgName="los-angeles-2016.jpg"
      link="/coming-soon"
      title="Los Angeles 2016"
    />
    <Tile
      imgName="graduation-2016.jpg"
      link="/coming-soon"
      title="Graduation 2016"
    />
  </div>
</div>
`;
